var/spack/repos/builtin/packages/margo/package.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Margo(AutotoolsPackage):
+    """Argobots bindings to the Mercury RPC API"""
+
+    homepage = "https://xgitlab.cels.anl.gov/sds/margo"
+    url      = "https://xgitlab.cels.anl.gov/sds/margo/-/archive/v0.4.3/margo-v0.4.3.tar.gz"
+    git      = "https://xgitlab.cels.anl.gov/sds/margo.git"
+
+    maintainers = ['fbudin69500', 'chuckatkins']
+
+    version('develop', branch='master')
+    version('0.4.3', sha256='61a634d6983bee2ffa06e1e2da4c541cb8f56ddd9dd9f8e04e8044fb38657475')
+
+    variant('shared', default=True,
+            description='Build shared libraries instead of static libraries')
+
+    depends_on('autoconf',  type='build')
+    depends_on('automake',  type='build')
+    depends_on('libtool',   type='build')
+    depends_on('m4',        type='build')
+    depends_on('pkgconfig', type='build')
+
+    depends_on('mercury')
+    depends_on('argobots@1.0rc1:')
```

Comment:
Again... I wouldn't assume that Spack understands the ordering of versions in `argobots`.  We need a way to just tell Spack what order versions are in (defining a complete ordering on a finite set of versions in a package should be pretty trivial).  Maybe considering changing `1.0rc1` to a numeric tuple?

Reply:
With the current implementation of the version comparison in spack, this actually works. It will also work for what I expect the next version to be: 1.0.0 as spack `version` class consider integer versions to be greater than alphabetical ones (`0` > `rc`).